### PR TITLE
Stop logical replication shard tracking in beforeIndexShardDeleted

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

See commits - just some minor improvements.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/io/crate/replication/logical/ShardReplicationService.java [hunk 1]
```diff
@@ -121,7 +121,7 @@ public void beforeIndexShardClosed(ShardId shardId,
     }
 
     @Override
-    public void afterIndexShardDeleted(ShardId shardId, Settings indexSettings) {
+    public void beforeIndexShardDeleted(ShardId shardId, Settings indexSettings) {
```
Comment (mfussenegger) [inline server/src/main/java/io/crate/replication/logical/ShardReplicationService.java:124]: I don't know if there was a good reason to have this on `after` ?

File: server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java [hunk 2]
```diff
@@ -361,8 +362,10 @@ private void renewLeasesThenReschedule(long toSeqNoReceived) {
                                     ThreadPool.Names.LOGICAL_REPLICATION,
                                     () -> renewLeasesThenReschedule(toSeqNoReceived)
                                 );
+                            } else if (isClosed) {
+                                LOGGER.info("Exception renewing retention lease. Stopping tracking (closed=true)");
```
Comment (seut) [inline server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java:366]: Do we want to log in this case? cause tracking should stop anyhow if closed, so an exception on retention lease renewal maybe of no interest?

File: server/src/main/java/io/crate/replication/logical/ShardReplicationService.java (hunk 1 above)
Comment (seut) [reply server/src/main/java/io/crate/replication/logical/ShardReplicationService.java:124]: Can't really remember, sorry. Maybe no real reason for it. Doing it `before` looks fine. What is the reason to change it? higher chance that the tracker (or request) actual stops when the shard is gone?

**Review by seut — APPROVED** [review]
👍 added some minor questions.

File: server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java (hunk 2 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java:366]: I'm fine with either - or maybe we put it on debug level?

Comment (mfussenegger) [reply server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java]: Changed it to debug level

**Pull request merged**